docs/markdown/Reference-manual.md

```diff
@@ -436,33 +436,46 @@ system) with the given name with `pkg-config` and [with
 CMake](Dependencies.md#cmake) if `pkg-config` fails. Additionally,
 frameworks (OSX only) and [library-specific fallback detection
 logic](Dependencies.md#dependencies-with-custom-lookup-functionality)
-are also supported. This function supports the following keyword
-arguments:
+are also supported.
+
+Dependencies can also be resolved by a fallback subproject.
+Versions of Meson before 0.55.0 will never automatically search for
+a fallback.  *Since 0.55.0* Meson can look for a wrap file or a directory
+with the same `dependency_name`; in that case, either the
+wrap file must have `dependency_name` in its `[provide]` section (see
+[Wrap documentation](Wrap-dependency-system-manual.md#provide-section)
+for more details), or the subproject must register the dependency using
+`meson.override_dependency('dependency_name', subproj_dep)`.
+This automatic search can be overridden using the `fallback` keyword
+argument.
+
+This function supports the following keyword arguments:
 
 - `default_options` *(since 0.37.0)*: an array of default option values
   that override those set in the subproject's `meson_options.txt`
   (like `default_options` in [`project()`](#project), they only have
   effect when Meson is run for the first time, and command line
   arguments override any default options in build files)
-- `fallback`: specifies a subproject fallback to use in case the
-  dependency is not found in the system. The value is an array
-  `['subproj_name', 'subproj_dep']` where the first value is the name
+- `fallback` (boolean argument, *since 0.56.0*): specifies whether Meson
+  should automatically pick a fallback subproject in case the dependency
+  is not found in the system.  By default, Meson will do so if `required` is
+  `true` or  [`enabled`](Build-options.md#features), or if a same-named
+  subproject has already been added with [`subproject()`](#subproject).
+- `fallback` (string or array argument): manually specifies a subproject
+  fallback to use in case the dependency is not found in the system.
+  This is useful if the automatic search is not applicable or if you
+  want to support older versions of Meson.  If the value is an array
```

Comment:
Instead of vague 'older' it should say 'older than 0.55.0'.

Reply:
We could even go a step further and deprecate this behavior.

Reply:
Deprecation of the string argument is probably a good idea. The array argument needs to stay in order to support older versions of Meson.